kernel/src/main/java/org/kframework/kdep/KDepModule.java

```diff
@@ -36,7 +38,8 @@ GlobalOptions globalOptions(KDepOptions options) {
     }
 
     @Provides
-    OuterParsingOptions outerParsingOptions(KDepOptions options) {
-        return options.outerParsing;
-    }
+    OuterParsingOptions outerParsingOptions(KDepOptions options) { return options.outerParsing; }
+
+    @Provides
+    OutputDirectoryOptions outputDirectoryOptions(KompileOptions options) { return options.outputDirectory; }
```

Comment:
It honestly doesn't matter. The only difference is that if it's request scoped, this function will be called only once instead of each time the options are requested. But this function is so simple it will be inlined by the jvm, so there's no significant difference in performance because of this.